autorest/adal/token_test.go

```diff
@@ -1083,6 +1083,12 @@ func TestNewMultiTenantServicePrincipalToken(t *testing.T) {
 	}
 }
 
+func TestMSIAvailable(t *testing.T) {
+	if MSIAvailable(context.Background(), http.DefaultClient) {
```

Comment:
Can we predicate the expected result based on some other well-known factor like an environment var?

Reply:
To see if you're in a MSI env, you can check if the MSI_ENDPOINT is set and, if not, you can check the IMDS endpoint availability. There are other variables that come into play to see which particular MSI env we're running in, but those two are the main ways to check for a MSI env.

Reply:
Let me clarify.  MSIAvailable() already checks the if the IMDS endpoint is available.  Running this test on your dev box, MSIAvailable() will return false, while running it in CI (Azure DevOps VMs) it returns true.

Given that this test is somewhat non-deterministic, maybe it's not a great candidate to run in CI.  I know we hav similar functionality in track 2, how do we test it there?

Reply:
We just need to mock the responses here, unclear why I overlooked that :p.